stock_dock/models/stock_dock.py

```diff
@@ -22,12 +23,11 @@ class StockDock(models.Model):
     company_id = fields.Many2one(
         comodel_name="res.company",
         string="Company",
-        related="warehouse_id.company_id",
-        readonly=True,
-        store=True,
-        index=True,
+        default=lambda self: self.env.company,
     )
 
     def _default_warehouse_id(self):
-        wh = self.env.ref("stock.warehouse0", raise_if_not_found=False)
+        wh = self.env["stock.warehouse"].search(
+            [("company_id", "=", self.env.company.id)]
+        )[0]
```

Comment:
Should be replaced with a limit = 1 instead of searching all warehouses to take a single one